ckanext/collaborators/logic/action.py

```diff
@@ -52,11 +51,10 @@ def dataset_collaborator_create(context, data_dict):
         raise toolkit.ObjectNotFound('User not found')
 
     toolkit.check_access('dataset_collaborator_create', context, data_dict)
-
     # Check if member already exists
     member = model.Session.query(DatasetMember).\
-        filter(DatasetMember.dataset_id == dataset.id).\
-        filter(DatasetMember.user_id == user.id).one_or_none()
+        filter(DatasetMember.dataset_id == dataset_id).\
```

Comment:
These are actually on purpose. `dataset_id` and `user_id` are the values provided by the user. These can be both the ids (UUID) or the names of the dataset / user. By using the id of the actual model objects we ensure we always use the UUIDs, which are the ones stored in the DB.